rero_mef/authorities/marctojson/do_idref_auth_person.py

```diff
@@ -0,0 +1,218 @@
+
+    @property
+    def json(self):
+        """Json data."""
+        return self.json_dict
+
+    def trans_idref_gender(self):
+        """Transformation gender 120 $a a:female, b: male, -:not known."""
+        if self.logger and self.verbose:
+            self.logger.info('Call Function', 'trans_gender')
+
+        gender = ""
+        fields_120 = self.marc.get_fields('120')
+        if fields_120 and fields_120[0] and fields_120[0].get_subfields('a'):
+            gender_type = fields_120[0].get_subfields('a')[0]
+            if gender_type == 'a':
+                gender = 'female'
+            elif gender_type == 'b':
+                gender = 'male'
+            elif gender_type == '-':
+                gender = 'not known'
+        if gender:
+            self.json_dict['gender'] = gender
+
+    def trans_idref_language_of_person(self):
+        """Transformation language_of_person 101 $a."""
+        if self.logger and self.verbose:
+            self.logger.info('Call Function', 'trans_language_of_person')
+        fields_101 = self.marc.get_fields('101')
+        language_list = []
+        if fields_101 and fields_101[0] and fields_101[0].get_subfields('a'):
+            for language in fields_101[0].get_subfields('a'):
+                language_list.append(language)
+        if language_list:
+            self.json_dict['language_of_person'] = language_list
+
+    def trans_pid(self):
+        """Transformation pid from field 001."""
+        if self.logger and self.verbose:
+            self.logger.info('Call Function', 'trans_pid')
+        fields_001 = self.marc.get_fields('001')
+        if fields_001 and fields_001[0]:
+            self.json_dict['pid'] = fields_001[0].data
+
+    def trans_idref_identifier_for_person(self):
+        """Transformation identifier_for_person from field 003."""
+        if self.logger and self.verbose:
+            self.logger.info('Call Function', 'trans_identifier_for_person')
+        fields_003 = self.marc.get_fields('003')
+        if fields_003 and fields_003[0]:
+            self.json_dict['identifier_for_person'] = fields_003[0].data
+
+    def trans_idref_birth_and_death_dates(self):
+        """Transformation birth_date and death_date."""
+        def format_103_date(date_str):
+            """Format date from 103.."""
+            date = ''
+            date_str = date_str.strip().replace(' ', '')
+            if date_str:
+                unknown = False
+                if date_str[-1] == '?':
+                    unknown = True
+                    date_str = date_str[:-1]
+                year = date_str[0:4]
+                month = date_str[4:6]
+                day = date_str[6:8]
+                if year:
+                    date = year
+                if month:
+                    date += '-' + month
+                if day:
+                    date += '-' + day
+                if unknown:
+                    date += '?'
+            return date or None
```

Comment:
as `date` is initialize to `''` (empty string), this method will never return None

Reply:
I think this is one way to change `''` to `None`. Python magic